docs/source/concepts/architectural-model.rst

```diff
@@ -0,0 +1,284 @@
+This may happen as part of command submission or for other reasons, e.g., load balancing.
+It is guaranteed that the ``EnteredViewEvent`` precedes contract usage, subject to the trust assumptions of the underlying ledger.
+
+A contract may enter and leave the view of a endpoint several times.
+For example, suppose that party `B` submits the following commands and their commits end up on the given domains.
+
+#. Create a contract `c` with stakeholders `A` and `B` on domain 2
+#. Exercise a non-consuming choice `ch1` on `c` on domain 1.
+#. Exercise a non-consuming choice `ch2` on `c` on domain 2.
+#. Exercise a consuming choice on `c` on domain 1.
+
+Then, the transaction tree stream that `P1` provides for `A` contains five events around contract `c`: ``EnteredViewEvent``, ``ExercisedEvent``, ``LeftViewEvent``, ``EnteredViewEvent``, ``ArchivedEvent``.
+These five events can be grouped into between two or five transactions.
+For example, the first three and the last two could be batched into one transaction each.
+However, `P1` cannot combine the ``LeftViewEvent`` with the subsequent ``EnteredViewEvent`` and `P1` must not elide them either.
+This is because their presence indicates that `P1`\ 's view may miss some events in between; in this example, exercising the choice `ch2`.
+
+The flat transaction stream by `P1` shows omits the non-consuming exercise choices.
+It nevertheless contains the three view change events ``EnteredViewEvent``, ``LeftViewEvent``, and ``EnteredViewEvent`` before the ``ArchivedEvent``.
+This is because the endpoint cannot know at the ``LeftViewEvent`` that there will be another ``EnteredViewEvent``.
+
+In contrast, `P2` need not output the ``EnteredViewEvent``\ s and ``LeftViewEvent``\ s at all in this example because `P2` hosts `A` on both domains.
+
+
+Relaxing the total ordering
+***************************
+
+The virtual shared ledger and the transaction service put all transactions into a linear sequence.
+Establishing such an order at all endpoints incurs synchronization, which hinders scalability.
+Different endpoints are therefore allowed to output events in different orders if these events do not causally depend on each other from the party's perspective.
+This section defines :ref:`causality <ledger-causality-party>` and illustrates the allowed and disallowed reorderings of commits by :ref:`examples <ledger-causality-examples>`.
+
+.. _ledger-causality-party:
+
+Causality
+=========
+
+Causality is defined on the commits in a party's projection of the virtual shared ledger.
+
+Definition »synchronizes with«
+  An action `act`\ :sub:`1` **synchronizes with** an action `act`\ :sub:`2` for party `A` if one of the following holds:
+
+  * **Create-use:**
+    `act`\ :sub:`1` creates the contract `c` and `act`\ :sub:`2` uses the contract, i.e., `act`\ :sub:`2` is a **Fetch** or an **Exercise** on `c` and `A` is a stakeholder of `c` and an informee of `act`\ :sub:`2`.
+ 
+  * **Use-use:**
```

Comment:
Does "use" include non-consuming actions? Do we really want two fetches to synchronize?